src/sources/http_server.rs

```diff
@@ -977,11 +1022,13 @@ mod tests {
             let mut headers = HeaderMap::new();
             headers.insert("User-Agent", "test_client".parse().unwrap());
             headers.insert("Upgrade-Insecure-Requests", "false".parse().unwrap());
+            headers.insert("X-Test-Header", "true".parse().unwrap());
```

Comment:
It'd be useful to have a separate unit test case (in addition to this) - where that tests the "*" results in all the headers being added.